ui-manchette-with-spacetimechart/src/hooks/useIsOverFlow.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useLayoutEffect, type RefObject } from 'react';
+
+type CallbackFunction = {
+  (hasOverflow: boolean): void;
+};
+
+export const useIsOverflow = (
+  ref: RefObject<HTMLElement>,
+  callback: CallbackFunction
+): boolean | undefined => {
+  const [isOverflow, setIsOverflow] = useState<boolean | undefined>(undefined);
```

Comment:
This could be a boolean all the time, it would make the code slightly more readable I think

Reply:
Sure !